pkg/component/runtime/conn_info_server.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package runtime
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	"github.com/elastic/elastic-agent/pkg/core/logger"
+)
+
+const (
+	defaultStopTimeout = 15 * time.Second
+	windows            = "windows"
+)
+
+type connInfoServer struct {
+	log         *logger.Logger
+	listener    net.Listener
+	waitCtx     context.Context
+	stopTimeout time.Duration
+}
+
+func newConnInfoServer(log *logger.Logger, comm Communicator, port int) (*connInfoServer, error) {
+	listener, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", port))
+	if err != nil {
+		return nil, fmt.Errorf("failed to start connection credentials listener: %w", err)
+	}
+
+	s := &connInfoServer{log: log, listener: listener, stopTimeout: defaultStopTimeout}
+
+	var cn context.CancelFunc
+	s.waitCtx, cn = context.WithCancel(context.Background())
+	go func() {
+		defer cn()
+		for {
+			conn, err := listener.Accept()
+			if err != nil {
+				log.Errorf("failed accept conn info connection: %v", err)
+				break
+			}
+			log.Debugf("client connected, sending connection info")
+			err = comm.WriteConnInfo(conn)
```

Comment:
Should we be doing any kind of verification of whose connection we accepted before handing out connection information?

Reply:
there is no additional verification in V1. This code implements exactly what V1 was doing, just refactored into a better encapsulated piece.

Reply:
Thanks, it may be worth coming back to this in the future but we can leave it as is since this is the way it currently works.